### PR TITLE
Hadoop 16455

### PR DESCRIPTION
**Pull request by bilaharith**
Implemented FileSystem.access() method for ABFS.
A new configuration has been added to switch on/off the functionality.

### CODE REVIEW

**Comment by bilaharith** [comment]
Driver test results using a Namespace enabled account in Central India:

<name>fs.azure.enable.check.access</name>
<value>false</value>

Tests run: 42, Failures: 0, Errors: 0, Skipped: 0
Tests run: 396, Failures: 0, Errors: 1, Skipped: 25
Tests run: 190, Failures: 0, Errors: 0, Skipped: 15

<name>fs.azure.enable.check.access</name>
<value>true</value>

Tests run: 42, Failures: 0, Errors: 0, Skipped: 0
Tests run: 396, Failures: 1, Errors: 1, Skipped: 21
Tests run: 190, Failures: 0, Errors: 0, Skipped: 15

File: hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AbfsConfiguration.java [hunk 1]
```diff
@@ -459,6 +463,10 @@ public boolean isUpnUsed() {
     return this.useUpn;
   }
 
+  public boolean isCheckAccessEnabled() {
+    return isCheckAccessEnabled;
```
Comment (snvijaya) [inline hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AbfsConfiguration.java]: Add this.

File: hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCheckAccess.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  @Test(expected = NullPointerException.class)
+  public void testCheckAccessWithNullPath() throws IOException {
+    Assume.assumeTrue(
+        ConfigurationKeys.FS_AZURE_ENABLE_CHECK_ACCESS + " is false",
+        getConfiguration().isCheckAccessEnabled());
+    getFileSystem().access(null, FsAction.READ);
+  }
+
+  @Test(expected = NullPointerException.class)
```
Comment (snvijaya) [inline hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCheckAccess.java]: Illegal Argument exception should be returned.

File: hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCheckAccess.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        getConfiguration().isCheckAccessEnabled());
+    FileSystem fs = getFileSystem();
+    Path testFilePath = new Path(TEST_FOLDER_PATH, "childfile1");
+    fs.create(testFilePath);
+    fs.access(testFilePath, null);
+  }
+
+  @Test
+  public void testCheckReadAccessForFileWithReadPermission()
+      throws IOException {
```
Comment (snvijaya) [inline hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCheckAccess.java]: Remove "throws IOException"

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 44 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1112 | trunk passed |
| +1 | compile | 38 | trunk passed |
| +1 | checkstyle | 22 | trunk passed |
| +1 | mvnsite | 31 | trunk passed |
| +1 | shadedclient | 710 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | trunk passed |
| 0 | spotbugs | 53 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 52 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 31 | the patch passed |
| +1 | compile | 22 | the patch passed |
| +1 | javac | 22 | the patch passed |
| -0 | checkstyle | 17 | hadoop-tools/hadoop-azure: The patch generated 2 new + 6 unchanged - 1 fixed = 8 total (was 7) |
| +1 | mvnsite | 26 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 712 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 18 | the patch passed |
| +1 | findbugs | 51 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 75 | hadoop-azure in the patch passed. |
| +1 | asflicense | 27 | The patch does not generate ASF License warnings. |
| | | 3081 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1213 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 11e359939dc5 4.4.0-157-generic #185-Ubuntu SMP Tue Jul 23 09:17:01 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3cc0ace |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/1/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/1/testReport/ |
| Max. process+thread count | 412 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 108 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1330 | trunk passed |
| +1 | compile | 30 | trunk passed |
| +1 | checkstyle | 22 | trunk passed |
| +1 | mvnsite | 35 | trunk passed |
| +1 | shadedclient | 804 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 24 | trunk passed |
| 0 | spotbugs | 60 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 58 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 35 | the patch passed |
| +1 | compile | 25 | the patch passed |
| +1 | javac | 25 | the patch passed |
| -0 | checkstyle | 17 | hadoop-tools/hadoop-azure: The patch generated 2 new + 6 unchanged - 1 fixed = 8 total (was 7) |
| +1 | mvnsite | 33 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 836 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | the patch passed |
| +1 | findbugs | 57 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 77 | hadoop-azure in the patch passed. |
| +1 | asflicense | 28 | The patch does not generate ASF License warnings. |
| | | 3626 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1213 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 58148fc5ba63 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 00b5a27 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/2/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/2/testReport/ |
| Max. process+thread count | 308 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 89 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1177 | trunk passed |
| +1 | compile | 28 | trunk passed |
| +1 | checkstyle | 20 | trunk passed |
| +1 | mvnsite | 32 | trunk passed |
| +1 | shadedclient | 742 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 25 | trunk passed |
| 0 | spotbugs | 55 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 52 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 27 | the patch passed |
| +1 | compile | 22 | the patch passed |
| +1 | javac | 22 | the patch passed |
| -0 | checkstyle | 14 | hadoop-tools/hadoop-azure: The patch generated 2 new + 6 unchanged - 1 fixed = 8 total (was 7) |
| +1 | mvnsite | 25 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 789 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | the patch passed |
| +1 | findbugs | 56 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 69 | hadoop-azure in the patch failed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 3299 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.fs.azurebfs.services.TestAbfsClientThrottlingAnalyzer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1213 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 65c3a2d71461 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9a1d8cf |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/3/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/3/artifact/out/patch-unit-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/3/testReport/ |
| Max. process+thread count | 400 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1254 | trunk passed |
| +1 | compile | 31 | trunk passed |
| +1 | checkstyle | 23 | trunk passed |
| +1 | mvnsite | 33 | trunk passed |
| +1 | shadedclient | 806 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | trunk passed |
| 0 | spotbugs | 49 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 47 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 26 | the patch passed |
| +1 | compile | 21 | the patch passed |
| +1 | javac | 21 | the patch passed |
| -0 | checkstyle | 15 | hadoop-tools/hadoop-azure: The patch generated 2 new + 6 unchanged - 1 fixed = 8 total (was 7) |
| +1 | mvnsite | 26 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 868 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 18 | the patch passed |
| +1 | findbugs | 62 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 87 | hadoop-azure in the patch passed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 3487 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1213 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux fe19224564ff 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 6244502 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/4/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/4/testReport/ |
| Max. process+thread count | 305 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 35 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1211 | trunk passed |
| +1 | compile | 26 | trunk passed |
| +1 | checkstyle | 19 | trunk passed |
| +1 | mvnsite | 29 | trunk passed |
| +1 | shadedclient | 782 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | trunk passed |
| 0 | spotbugs | 56 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 53 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 27 | the patch passed |
| +1 | compile | 23 | the patch passed |
| +1 | javac | 23 | the patch passed |
| -0 | checkstyle | 18 | hadoop-tools/hadoop-azure: The patch generated 2 new + 6 unchanged - 1 fixed = 8 total (was 7) |
| +1 | mvnsite | 26 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 830 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 24 | the patch passed |
| +1 | findbugs | 61 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 88 | hadoop-azure in the patch passed. |
| +1 | asflicense | 27 | The patch does not generate ASF License warnings. |
| | | 3392 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1213 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ddc2a45bd012 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 69ddb36 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/5/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/5/testReport/ |
| Max. process+thread count | 306 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 76 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1260 | trunk passed |
| +1 | compile | 31 | trunk passed |
| +1 | checkstyle | 26 | trunk passed |
| +1 | mvnsite | 34 | trunk passed |
| +1 | shadedclient | 840 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 23 | trunk passed |
| 0 | spotbugs | 52 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 50 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 26 | the patch passed |
| +1 | compile | 22 | the patch passed |
| +1 | javac | 22 | the patch passed |
| -0 | checkstyle | 17 | hadoop-tools/hadoop-azure: The patch generated 2 new + 6 unchanged - 1 fixed = 8 total (was 7) |
| +1 | mvnsite | 26 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 834 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 24 | the patch passed |
| +1 | findbugs | 66 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 90 | hadoop-azure in the patch passed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 3559 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1213 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 74620992d9cf 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3329257 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/6/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/6/testReport/ |
| Max. process+thread count | 306 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 44 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1300 | trunk passed |
| +1 | compile | 31 | trunk passed |
| +1 | checkstyle | 23 | trunk passed |
| +1 | mvnsite | 33 | trunk passed |
| +1 | shadedclient | 854 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 25 | trunk passed |
| 0 | spotbugs | 56 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 54 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 31 | the patch passed |
| +1 | compile | 26 | the patch passed |
| +1 | javac | 26 | the patch passed |
| -0 | checkstyle | 18 | hadoop-tools/hadoop-azure: The patch generated 2 new + 6 unchanged - 1 fixed = 8 total (was 7) |
| +1 | mvnsite | 30 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 893 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 23 | the patch passed |
| +1 | findbugs | 63 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 81 | hadoop-azure in the patch passed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 3667 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1213 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 0928be674a93 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 915cbc9 |
| Default Java | 1.8.0_222 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/7/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/7/testReport/ |
| Max. process+thread count | 415 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1213/7/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**